projects/js-packages/publicize-components/src/components/social-image-generator/template-picker/index.js

```diff
@@ -16,31 +15,23 @@ import TEMPLATES_DATA from './templates.js';
  * @returns {React.ReactNode} - The component's rendered output.
  */
 const TemplatePicker = ( { value = null, onTemplateSelected = null } ) => {
-	const [ selectedTemplate, setSelectedTemplate ] = useState( value );
-
-	// Add memoized callback function for each template.
-	const TEMPLATES = useMemo(
-		() =>
-			TEMPLATES_DATA.map( template => ( {
-				...template,
-				onSelect: () => {
-					setSelectedTemplate( template.name );
-					if ( onTemplateSelected ) {
-						onTemplateSelected( template.name );
-					}
-				},
-			} ) ),
-		[ setSelectedTemplate, onTemplateSelected ]
+	const onTemplateClicked = useCallback(
+		event => {
+			const templateName = event.target.id;
```

Comment:
Are we using a const to make it explicit that we're looking for the template's name?

Reply:
I think it makes it more readable, than just `onTemplateSelected?.( event.target.id );`

Reply:
I don't have strong feelings really. I was more curious. It's fine to leave it as is.